### PR TITLE
Po 72

### PR DESCRIPTION
**Pull request by max-holland**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/PO-72

### Change description ###
Added call to back end SSO endpoints

### Checklist

- [X] commit messages are meaningful and follow good commit message guidelines
- [] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: server/routes.ts [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
+  if (ssoEnabled) {
+    router.get('/sso/login', (req, res) => ssoLogin(req, res));
+    router.post('/sso/login-callback', (req, res, next) => ssoLoginCallback(req, res, next));
+    router.get('/sso/logout', (req, res, next) => ssoLogout(req, res, next));
+    router.get('/sso/logout-callback', (req, res, next) => ssoLogoutCallback(req, res, next));
+    router.get('/sso/authenticated', (req, res) => ssoAuthenticated(req, res));
+  } else {
+    router.get('/sso/login', (req, res) => ssoLoginStub(res));
+    router.get('/sso/login-callback', (req, res, next) => ssoLoginCallbackStub(req, res, next));
+    router.get('/sso/authenticated', (req, res) => ssoAuthenticatedStub(req, res));
```
[the remark this answers is not in the document]
Comment (max-holland) [reply server/routes.ts:35]: Rate limit not requirement. Will address later as per @danlysiak.

File: server/routes.ts [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
-  router.get('/sso/authenticated', (req, res) => ssoAuthenticated(req, res));
+  if (ssoEnabled) {
+    router.get('/sso/login', (req, res) => ssoLogin(req, res));
+    router.post('/sso/login-callback', (req, res, next) => ssoLoginCallback(req, res, next));
+    router.get('/sso/logout', (req, res, next) => ssoLogout(req, res, next));
+    router.get('/sso/logout-callback', (req, res, next) => ssoLogoutCallback(req, res, next));
+    router.get('/sso/authenticated', (req, res) => ssoAuthenticated(req, res));
+  } else {
+    router.get('/sso/login', (req, res) => ssoLoginStub(res));
+    router.get('/sso/login-callback', (req, res, next) => ssoLoginCallbackStub(req, res, next));
```
[the remark this answers is not in the document]
Comment (max-holland) [reply server/routes.ts:34]: Rate limit not requirement. Will address later as per @danlysiak.

File: server/routes.ts [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
 
   // Handle our authentication
-  router.get('/sso/login', (req, res, next) => ssoLogin(req, res, next));
-  router.get('/sso/login-callback', (req, res, next) => ssoLoginCallback(req, res, next));
-  router.get('/sso/logout', (req, res, next) => ssoLogout(req, res, next));
-  router.get('/sso/logout-callback', (req, res, next) => ssoLogoutCallback(req, res, next));
-  router.get('/sso/authenticated', (req, res) => ssoAuthenticated(req, res));
+  if (ssoEnabled) {
+    router.get('/sso/login', (req, res) => ssoLogin(req, res));
+    router.post('/sso/login-callback', (req, res, next) => ssoLoginCallback(req, res, next));
```
[the remark this answers is not in the document]
Comment (max-holland) [reply server/routes.ts:28]: Rate limit not requirement. Will address later as per @danlysiak.

File: server/routes.ts [hunk 4]
```diff
@@ -1,19 +1,41 @@
...
   router.use('/api', proxy());
 
   // Handle our authentication
-  router.get('/sso/login', (req, res, next) => ssoLogin(req, res, next));
-  router.get('/sso/login-callback', (req, res, next) => ssoLoginCallback(req, res, next));
-  router.get('/sso/logout', (req, res, next) => ssoLogout(req, res, next));
-  router.get('/sso/logout-callback', (req, res, next) => ssoLogoutCallback(req, res, next));
-  router.get('/sso/authenticated', (req, res) => ssoAuthenticated(req, res));
+  if (ssoEnabled) {
+    router.get('/sso/login', (req, res) => ssoLogin(req, res));
```
[the remark this answers is not in the document]
Comment (max-holland) [reply server/routes.ts:27]: Rate limit not requirement. Will address later as per @danlysiak.

File: server/routes.ts [hunk 5]
```diff
@@ -1,19 +1,41 @@
...
+    router.post('/sso/login-callback', (req, res, next) => ssoLoginCallback(req, res, next));
+    router.get('/sso/logout', (req, res, next) => ssoLogout(req, res, next));
+    router.get('/sso/logout-callback', (req, res, next) => ssoLogoutCallback(req, res, next));
+    router.get('/sso/authenticated', (req, res) => ssoAuthenticated(req, res));
```
[the remark this answers is not in the document]
Comment (max-holland) [reply server/routes.ts:31]: Rate limit not requirement. Will address later as per @danlysiak.

File: server/routes.ts [hunk 6]
```diff
@@ -1,19 +1,48 @@
...
+
+  router.get('/sso/login', (req: Request, res: Response) => login(req, res));
+  router[loginCallbackType]('/sso/login-callback', (req: Request, res: Response, next: NextFunction) =>
+    loginCallback(req, res, next),
+  );
+  router.get('/sso/logout', (req: Request, res: Response, next: NextFunction) => logout(req, res, next));
+  router.get('/sso/logout-callback', (req: Request, res: Response, next: NextFunction) =>
+    logoutCallback(req, res, next),
+  );
+  router.get('/sso/authenticated', (req: Request, res: Response) => authenticated(req, res));
```
[the remark this answers is not in the document]
Comment (max-holland) [reply server/routes.ts:33]: Not current requirement. Will address later as per @danlysiak

**Review by sabahirfan — APPROVED** [review]

**Pull request merged**